Doc/whatsnew/3.14.rst

```diff
@@ -1027,6 +1027,12 @@ New features
 * Add :c:func:`PyUnstable_Object_EnableDeferredRefcount` for enabling
   deferred reference counting, as outlined in :pep:`703`.
 
+* The :ref:`Unicode Exception Objects <unicodeexceptions>` C API now
+  type-checks its exception argument, raising a :exc:`TypeError` if
+  the latter is not an instance of :exc:`UnicodeError`.
```

Comment:
The wording here is a little confusing. I think it's fine to just say something simpler, like "The Unicode Exception Objects C API now raises a `TypeError` if it's given an exception that isn't a `UnicodeError`"